PWGHF/HFC/TableProducer/correlatorDstarHadrons.cxx

```diff
@@ -0,0 +1,293 @@
+  Configurable<float> dcaxyAssoTrackMax{"dcaxyAssoTrackMax", 10.0, "max DCAxy of Associated Track"};
+  Configurable<float> dcazAssoTrackMin{"dcazAssoTrackMin", 0.0, "min DCAz of Associated Track"};
+  Configurable<float> dcazAssoTrackMax{"dcazAssoTrackMax", 10.0, "max DCAz of Associated Track"};
+  Configurable<float> ptAssoTrackMin{"ptAssoTrackMin", 0.5, "min Pt of Associated Track"};
+  Configurable<float> ptAssoTrackMax{"ptAssoTrackMax", 50.0, "max pT of Associated Track"};
+
+  ConfigurableAxis binsMultiplicity{"binsMultiplicity", {VARIABLE_WIDTH, 0.0f, 2000.0f, 6000.0f, 100000.0f}, "Mixing bins - multiplicity"};
+  ConfigurableAxis binsZVtx{"binsZVtx", {VARIABLE_WIDTH, -10.0f, -2.5f, 2.5f, 10.0f}, "Mixing bins - z-vertex"};
+
+  // ColumnBinningPolicy<aod::collision::PosZ, aod::mult::MultFV0M<aod::mult::MultFV0A, aod::mult::MultFV0C>> binningScheme{{binsZVtx, binsMultiplicity},true};
+  // ColumnBinningPolicy<aod::collision::PosZ, aod::mult::MultFT0M<aod::mult::MultFT0A, aod::mult::MultFT0C>> binningScheme;
+  using BinningType = ColumnBinningPolicy<aod::collision::PosZ, aod::mult::MultFV0M<aod::mult::MultFV0A, aod::mult::MultFV0C>>;
+  BinningType binningScheme{{binsZVtx, binsMultiplicity}, true};
+
+  // Collision Table
+  using CollisionsWDstar = soa::Join<aod::Collisions, aod::Mults, aod::DmesonSelection>;
+  using FilteredCollisions = soa::Filtered<CollisionsWDstar>;
+
+  // candidate table
+  using DstarCandidates = soa::Join<aod::HfCandDstars, aod::HfSelDstarToD0Pi>; // Added extra cloumns in HfCandDstar (Prong0Id, Prong1Id), so no need to add table HfCandD0Fromdstar
+  using FilteredCandidates = soa::Filtered<DstarCandidates>;
+
+  using FilteredTracks = soa::Filtered<aod::TracksWDca>;
+
+  // collision table filter
+  Filter collisionFilter = aod::hf_selection_dmeson_collision::dmesonSel == selectOnlyCollisionWDstar;
+  // candidate filter
+  Filter candidateFilter = aod::hf_sel_candidate_dstar::isSelDstarToD0Pi == selectionFlagDstar;
+  // track table filter
+  Filter trackFilter = nabs(aod::track::eta) <= etaAbsAssoTrackMax && aod::track::pt >= ptAssoTrackMin && aod::track::pt <= ptAssoTrackMax &&
+                       aod::track::dcaXY >= dcaxyAssoTrackMin && aod::track::dcaXY <= dcaxyAssoTrackMax &&
+                       aod::track::dcaZ >= dcazAssoTrackMin && aod::track::dcaZ <= dcazAssoTrackMax;
+  SliceCache cache;
+  // Preslice<DstarCandidates> perColCandidates = aod::hf_cand::collisionId;
+  Preslice<FilteredCandidates> perColCandidates = aod::hf_cand::collisionId;
+  // Preslice<aod::TracksWDca> perColTracks = aod::track::collisionId;
+  Preslice<FilteredTracks> perColTracks = aod::track::collisionId;
+
+  HistogramRegistry registry{
+    "registry",
+    {{"hTriggerColCandPairCounts", "Counts of Trigger Collision, Trigger Candidates and Pair Counts", {HistType::kTH1F, {{3, 0.0, 3.0}}}}}};
+
+  void init(InitContext&)
+  {
+    binningScheme = {{binsZVtx, binsMultiplicity}, true};
+  }
```

Comment:
Whenever you have multiple process functions that are mutually exclusive, check the process functions in `init` and throw a fatal error if they are not set correctly.

Reply:
done, see if it looks good, or if we need something else.

Reply:
The check should be the first thing done in `init` so that you don't perform other operations unnecessarily.

Reply:
done